### DISCUSSION TITLE
How do you maintain the plugins in your YOURLS install?

### DISCUSSION THREAD

**Discussion by jfix** [YOURLS/YOURLS#3016] [Plugins]
Hi, the following is more of a thought experiment. :-)

I wonder how people manage plugins within their YOURLS installations.

In my case, for example, I find it quite troublesome.  I have two environments that use the same plugins. Some of these plugins are my own, some come from elsewhere. The code base of my customised YOURLS projects are hosted with GitHub. When I want to use a new plugin that I find, say, via the `awsome-yourls` repository, I will have to clone the project into my own code base or download the zip file and add it to my project. The external code needs to be committed in my own repository which will trigger a deploy via GitHub actions. The plugin will then need to be activated manually via the plugins management page. When there is an update (if I find out about it, probably by chance only) I have to update my cloned version somehow. The process, to be honestly, feels a bit clumsy.

I guess what I'm wondering is whether there is some thoughts given to improve the handling of plugins with YOURLS?  Maybe "improving" is not the right word because there may be an efficient way to manage plugins, and I'm just not aware of it. In that case, I'd be happy to hear about it.

Otherwise, I wonder whether some sort of Composer-style or Wordpress plugin-style approach might be envisaged? Maybe a search interface that would allow to discover plugins (via the `awsome-yourls` repo?), some alerting mechanism regarding new versions and of course installation and removal via that user interface.

Probably, surely, it's overkill for YOURLS. :-) But it's summer and the right time for bouncing ideas like this one around  ...

**Comment by ozh** [comment]
There is an updater plugin I think in the list

Another way could be to install plugins from the sources via git and recursively git pull in all the plugin directories.

As for composer : I started https://github.com/YOURLS/composer-installer but I think it doesn't work with composer 2
